src/utils.js

```diff
@@ -335,14 +335,31 @@ ko.utils = (function () {
             return element && element.tagName && element.tagName.toLowerCase();
         },
 
+        catchFunctionErrors: function (delegate) {
+            return ko['onError'] ? function () {
+                try {
+                    return delegate.apply(this, arguments);
+                } catch (e) {
+                    ko['onError'] && ko['onError'](e);
+                    throw e;
+                }
+            } : delegate;
+        },
+
+        setTimeout: function (handler, timeout) {
+            return setTimeout(ko.utils.catchFunctionErrors(handler), timeout);
```

Comment:
Would this be better as a mutation observer, animation frame, deferral to, or refactor of `ko.tasks` [scheduler](https://github.com/knockout/knockout/blob/7d263dcbca18eba03cc7dca5f5f73d354aa77a31/src/tasks.js#L7)? That would feel more consistent – is there logic in keeping it separate?

Perhaps, if so, then this should not be `ko.utils.setTimeout` but rather `ko.utils.runAsync` (or something similar).

Purely thoughts & suggestions, of course. :)